staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -62,6 +62,42 @@ var (
 		},
 		[]string{"resource"},
 	)
+	watchersCounter = metrics.NewCounterVec(
+		&metrics.CounterOpts{
+			Name: "apiserver_created_watchers",
+			Help: "Counter of watchers created in watchcache broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	watchersGauge = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Name: "apiserver_realtime_watchers",
+			Help: "Gauge of watchers watching in watchcache broken by resource type",
+		},
+		[]string{"resource"},
+	)
+	cacherEventsCounter = metrics.NewCounterVec(
```

Comment:
Isn't that duplication of: https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go#L163
?

Reply:
i can't remember that for sure, but the HighWaterMark thing was added for a reason instead of metric.
I think that was performance reason.

I'm reluctant for exposing that as a metric directly - maybe we can do something like exposing it as a metric, but not on every call, but once per second or something like that?